syntax = "proto3";
package proxy_service;

import public "google/protobuf/empty.proto";

option java_multiple_files = true;
option java_package = "com.cinch.grpc";

message RegisterAgentRequest {
    string hostname = 1;
}

message RegisterAgentResponse {
    int64 agent_id = 1;
    string proxy_url = 2;
}

message RegisterPathRequest {
    int64 agent_id = 1;
    string path = 2;
}

message RegisterPathResponse {
    int64 path_id = 1;
}

message AgentInfo {
    int64 agent_id = 1;
}

message ScrapeRequest {
    int64 agent_id = 1;
    int64 scrape_id = 2;
    string path = 3;
}

message ScrapeResponse {
    int64 agent_id = 1;
    int64 scrape_id = 2;
    bool valid = 3;
    int32 status_code = 4;
    string text = 5;
}

service ProxyService {

    rpc registerAgent (RegisterAgentRequest) returns (RegisterAgentResponse) {
    }

    rpc registerPath (RegisterPathRequest) returns (RegisterPathResponse) {
    }

    rpc readRequestsFromProxy (AgentInfo) returns (stream ScrapeRequest) {
    }

    rpc writeResponseToProxy (ScrapeResponse) returns (google.protobuf.Empty) {
    }
}

